### DISCUSSION TITLE
how to add onload event listener to links

### DISCUSSION THREAD

**Discussion by lili21** [remix-run/remix#3856] [Q&A]
```js
export const links = () => [
  {
    rel: 'preload',
    href: styleUrl,
    as: 'style',
    onload: () => conosle.log('loaded')
  }
]
```

it doesn't work.

**Comment by kiliman** [comment]
Not supported. Those resources will most likely be loaded before Remix starts anyway.

**Reply by lili21** [reply]
I want to lazy load some non-critical css.  I used to do it by this way

```html
<link rel="preload" href="index.css" as="style" onload="this.onload=null;this.rel='stylesheet'" />
```

**Reply by sergiodxa** [reply]
You could add two link tags, one for preload and another with rel stylesheet, that should unblock rendering as the styles are loaded

**Reply by lili21** [reply]
you mean something like this ?

```html
<link rel="preload" href="https://unpkg.com/@arco-design/web-react@2.38.0/dist/css/arco.css" as="style">
<link rel="stylesheet" href="https://unpkg.com/@arco-design/web-react@2.38.0/dist/css/arco.css">
```

it still block rendering.

**Comment by tylerscave** [comment]
Also curious about this... Any other ideas?

**Comment by gitaugakwa** [comment]
Any update on this?

**Reply by sergiodxa** [reply]
With React v19 LinksFunction is most likely not going to be needed anymore as you can render <link> tags anywhere and React will move them to the head now.

**Comment by ben-wfp** [comment]
I have a little pattern which works for me. First add a data attribute to the link tag like

`<link rel="stylesheet" href="/styles.css" media="all" data-async-style="true">`

then update the links in an effect, applying the styles:

```
  useEffect(() => {
    const asyncLinks = document.querySelectorAll('link[data-async-style]');

    asyncLinks.forEach(link => link.setAttribute('media', 'all'));
  }, []);
```
